commit 248a2f3cbebe36f1ea2ac32f1e58d2d57f9f9fdd
Author: Chen-Yu Tsai <wens@csie.org>
Date:   Tue Feb 5 23:42:24 2019 +0800

    mmc: sunxi: Filter out unsupported modes declared in the device tree
    
    commit d6f11e7d91f2ac85f66194fe3ef8789b49901d64 upstream.
    
    The MMC device tree bindings include properties used to signal various
    signalling speed modes. Until now the sunxi driver was accepting them
    without any further filtering, while the sunxi device trees were not
    actually using them.
    
    Since some of the H5 boards can not run at higher speed modes stably,
    we are resorting to declaring the higher speed modes per-board.
    
    Regardless, having boards declare modes and blindly following them,
    even without proper support in the driver, is generally a bad thing.
    
    Filter out all unsupported modes from the capabilities mask after
    the device tree properties have been parsed.
    
    Cc: <stable@vger.kernel.org>
    Signed-off-by: Chen-Yu Tsai <wens@csie.org>
    Acked-by: Maxime Ripard <maxime.ripard@bootlin.com>
    Signed-off-by: Ulf Hansson <ulf.hansson@linaro.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/mmc/host/sunxi-mmc.c b/drivers/mmc/host/sunxi-mmc.c
index 568349e1fbc2..c4584184525f 100644
--- a/drivers/mmc/host/sunxi-mmc.c
+++ b/drivers/mmc/host/sunxi-mmc.c
@@ -1394,6 +1394,21 @@ static int sunxi_mmc_probe(struct platform_device *pdev)
 	if (ret)
 		goto error_free_dma;
 
+	/*
+	 * If we don't support delay chains in the SoC, we can't use any
+	 * of the higher speed modes. Mask them out in case the device
+	 * tree specifies the properties for them, which gets added to
+	 * the caps by mmc_of_parse() above.
+	 */
+	if (!(host->cfg->clk_delays || host->use_new_timings)) {
+		mmc->caps &= ~(MMC_CAP_3_3V_DDR | MMC_CAP_1_8V_DDR |
+			       MMC_CAP_1_2V_DDR | MMC_CAP_UHS);
+		mmc->caps2 &= ~MMC_CAP2_HS200;
+	}
+
+	/* TODO: This driver doesn't support HS400 mode yet */
+	mmc->caps2 &= ~MMC_CAP2_HS400;
+
 	ret = sunxi_mmc_init_host(host);
 	if (ret)
 		goto error_free_dma;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1417_248a2f3cbebe36f1ea2ac32f1e58d2d57f9f9fdd.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1417_248a2f3cbebe36f1ea2ac32f1e58d2d57f9f9fdd.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
